modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/notifications/SoapServiceNotificationAndSubscriptionTestCase.java

```diff
@@ -0,0 +1,250 @@
+    }
+
+    @Test(groups = {"wso2.greg", "wso2.greg.es"}, description = "Adding wrong subscription method to check the error message")
+    public void addWrongSubscriptionMethod() throws JSONException, IOException {
+
+        JSONObject dataObject = new JSONObject();
+
+        dataObject.put("notificationType","PublisherCheckListItemUnchecked");
+        dataObject.put("notificationMethod","test");
+
+        ClientResponse response =
+                genericRestClient.geneticRestRequestPost(publisherUrl + "/subscriptions/soapservice/" + assetId, MediaType.APPLICATION_JSON,
+                        MediaType.APPLICATION_JSON, dataObject.toString()
+                        , queryParamMap, headerMap, cookieHeader);
+
+        String payLoad = response.getEntity(String.class);
+        payLoad = payLoad.substring(payLoad.indexOf('{'));
+        JSONObject obj = new JSONObject(payLoad);
+        assertNotNull(obj.get("error").toString(),
+                "Error message is not contained in the response for notification method \"test\"" + response
+                        .getEntity(String.class));
+    }
+
+    private void deleteSoapServiceAsset() throws JSONException {
+        genericRestClient.geneticRestRequestDelete(publisherUrl + "/assets/" + assetId, MediaType.APPLICATION_JSON,
+                MediaType.APPLICATION_JSON, queryParamMap, headerMap, cookieHeader);
+    }
+
+    private void setTestEnvironment() throws JSONException, IOException {
+        // Authenticate
+        ClientResponse response = genericRestClient
+                .geneticRestRequestPost(publisherUrl + "/authenticate/", MediaType.APPLICATION_FORM_URLENCODED,
+                        MediaType.APPLICATION_JSON, "username=admin&password=admin", queryParamMap, headerMap, null);
+        JSONObject obj = new JSONObject(response.getEntity(String.class));
+        jSessionId = obj.getJSONObject("data").getString("sessionId");
+        cookieHeader = "JSESSIONID=" + jSessionId;
+
+        //Create rest service
+        queryParamMap.put("type", "soapservice");
+        String dataBody = readFile(resourcePath + "json" + File.separator + "publisherPublishSoapResource.json");
+        ClientResponse createResponse = genericRestClient
+                .geneticRestRequestPost(publisherUrl + "/assets", MediaType.APPLICATION_JSON,
+                        MediaType.APPLICATION_JSON, dataBody, queryParamMap, headerMap, cookieHeader);
+        JSONObject createObj = new JSONObject(createResponse.getEntity(String.class));
+        assetId = createObj.get("id").toString();
+    }
+
+    @AfterClass
+    public void clean() throws Exception {
+        deleteSoapServiceAsset();
+    }
+
+    @DataProvider
+    private static TestUserMode[][] userModeProvider() {
+        return new TestUserMode[][]{
+                new TestUserMode[]{TestUserMode.SUPER_TENANT_ADMIN}
+                //                new TestUserMode[]{TestUserMode.TENANT_USER},
```

Comment:
This is used to run tests in tenant mode, but due to existing issues in running tests in tenant mode this line has been commented out